libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigator.java

```diff
@@ -37,6 +38,22 @@ void updateLocation(Location raw) {
     }
   }
 
+  // TODO this call should be done in the background - it's currently blocking the UI
```

Comment:
Should we be handling this with an `AsyncTask` for example?  Or should we note for developers that calling it will block the UI thread.  I guess we can offer a callback with option 1.

Reply:
Yeah, that's a great point and I agree, we should handle it. For the sake of getting this PR moving what about ticketing this `TODO` out and merging here?

Reply:
@Guardiola31337 yeah if we don't want to address this in this PR, the javadoc should be updated to make it clear that this will block the UI thread when called.